mpas_analysis/ocean/meridional_heat_transport.py

```diff
@@ -0,0 +1,213 @@
+from ..shared.climatology import climatology
+
+
+def meridional_heat_transport(config):  # {{{
+    """
+    Process MHT analysis member data if available.
+    Plots MHT as:
+       1D function of latitude
+       2D function of latitude and depth
+
+    Parameters
+    ----------
+    config : instance of MpasAnalysisConfigParser
+        configuration options used to customize the analysis task
+
+    Authors
+    -------
+    Mark Petersen
+
+    Last Modified
+    -------------
+    04/21/2017
+    """
+
+    # **** Initial settings ****
+
+    # perform common setup for the task
+    namelist, runStreams, historyStreams, calendar, namelistMap, streamMap, \
+        variableMap, plotsDirectory = setup_task(config, componentName='ocean')
+
+    check_analysis_enabled(
+        namelist=namelist,
+        analysisOptionName='config_am_timeseriesstatsmonthly_enable',
+        namelistMap=namelistMap,
+        raiseException=True)
+    check_analysis_enabled(
+        namelist=namelist,
+        analysisOptionName='config_am_meridionalheattransport_enable',
+        namelistMap=namelistMap,
+        raiseException=False)
+
+    # Get a list of timeSeriesStats output files from the streams file,
+    # reading only those that are between the start and end dates
+    #   First a list necessary for theMHT climatology
+    streamName = historyStreams.find_stream(streamMap['timeSeriesStats'])
+    startDateClimo = config.get('climatology', 'startDate')
+    endDateClimo = config.get('climatology', 'endDate')
+    inputFilesClimo = historyStreams.readpath(streamName,
+                                              startDate=startDateClimo,
+                                              endDate=endDateClimo,
+                                              calendar=calendar)
+    simulationStartTime = get_simulation_start_time(runStreams)
+
+    print '\n  List of files for climatologies:\n' \
+          '    {} through\n    {}'.format(
+              os.path.basename(inputFilesClimo[0]),
+              os.path.basename(inputFilesClimo[-1]))
+
+    startYearClimo = config.getint('climatology', 'startYear')
+    endYearClimo = config.getint('climatology', 'endYear')
+    #   Create dictionary to store Climo related variables
+    dictClimo = {'inputFilesClimo': inputFilesClimo,
+                 'startDateClimo': startDateClimo,
+                 'endDateClimo':  endDateClimo,
+                 'startYearClimo': startYearClimo,
+                 'endYearClimo': endYearClimo}
+
+    sectionName = 'meridionalHeatTransport'
+
+    # Read in depth and MHT latitude points
+    # Latitude is from binBoundaryMerHeatTrans written in
+    #  mpaso.hist.am.meridionalHeatTransport.*.nc
+    # Depth is from refZMid, also in mpaso.hist.am.meridionalHeatTransport.*.nc
+    try:
+        mhtFile = historyStreams.readpath('meridionalHeatTransportOutput')[0]
+    except ValueError:
+        raise IOError('No MPAS-O MHT history file found: need at least one ')
```

Comment:
Actually, I think the better option would be to add a check to make sure the analysis member is enabled in the namelist and to return (rather than causing an error) if it isn't.  It's not meant to be an error to not have a given AM enabled in a given run.